File: Hello World/helloworld.c
#include <stdio.h>
#include <string.h> 
#include <mpi.h>
#define N 20

int main(int argc, char **argv)
{
  char mensaje[N];
  int  i, my_id, nproc;
  MPI_Status status;
  int root = 0;
  double start_time, end_time, elapsed_time;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nproc);
  MPI_Comm_rank(MPI_COMM_WORLD, &my_id);

  // Inicio del temporizador
  start_time = MPI_Wtime();

  if (my_id == root)
  {
    strcpy(mensaje, "Hola mundo");
  }
  MPI_Bcast(mensaje, N, MPI_CHAR, root, MPI_COMM_WORLD);
  printf("Mensaje del proceso %d : %s\n", my_id, mensaje);

  end_time = MPI_Wtime();

	elapsed_time = end_time - start_time;

	double max_time;
  MPI_Reduce(&elapsed_time, &max_time, 1, MPI_DOUBLE, MPI_MAX, 0, MPI_COMM_WORLD);

	if (my_id == 0) {
	  printf("Tiempo total de ejecución = %f segundos\n", max_time);
  }

  MPI_Finalize();
}

File: Hello World/helloworld.py
from mpi4py import MPI
from decimal import Decimal

N = 20
mensaje = bytearray(b' ' * N)  # Se inicializa el mensaje como un bytearray de tamaño N
root = 0

start_time = MPI.Wtime()
comm = MPI.COMM_WORLD
my_id = comm.Get_rank()
nproc = comm.Get_size()

if my_id == root:
    mensaje[:len("Hola mundo")] = b"Hola mundo"  # Copiar "Hola mundo" en el bytearray

# Transmitir el mensaje desde el proceso raíz
comm.Bcast(mensaje, root=root)

# Convertir el mensaje a una cadena para imprimir
print(f"Mensaje del proceso {my_id}: {mensaje.decode('utf-8').strip()}")

end_time = MPI.Wtime()
execution_time = end_time - start_time
total_execution_time = comm.reduce(execution_time, op=MPI.MAX, root=0)

# El proceso raíz (0) imprime el tiempo total de ejecución
if my_id == 0:
    print(f"Tiempo total de ejecución: {total_execution_time} segundos")